lightning/src/chain/channelmonitor.rs

```diff
@@ -1140,7 +1147,7 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 		mem::swap(&mut new_holder_commitment_tx, &mut self.current_holder_commitment_tx);
 		self.prev_holder_signed_commitment_tx = Some(new_holder_commitment_tx);
 		if self.holder_tx_signed {
-			return Err(MonitorUpdateError("Latest holder commitment signed has already been signed, update is rejected"));
+			return Err(MonitorUpdateError::NoPersistMonitor("Latest holder commitment signed has already been signed, update is rejected"));
```

Comment:
I think we need to persist anyway on this returrn otherwise you will break concurrent monitoring in case of distributed setup. If you don't store this update and fail-and-restore this ChannelMonitor instance it won't claim concurrent broadcast for an ulterior state broadcast by another ChanneMonitor instance.

But we should still return an error to the `update_monitor` caller. IMO, persistence-acceptance is different from state-acceptance.

Reply:
Ahh, that was the whole point of adding `MonitorUpdateErr::Persist/NoPersist`, thanks.